inc/functions.php

```diff
@@ -6849,13 +6889,36 @@ function fetch_remote_file($url, $post_data=array())
 		{
 			return false;
 		}
+
+		$data = null;
+
 		while(!feof($fp))
 		{
 			$data .= fgets($fp, 12800);
 		}
 		fclose($fp);
+
 		$data = explode("\r\n\r\n", $data, 2);
-		return $data[1];
+
+		$header = $data[0];
+		$status_line = current(explode("\n\n", $header, 1));
+		$body = $data[1];
+
+		if($max_redirects != 0 && (strstr($status_line, ' 301 ') || strstr($status_line, ' 302 ')))
+		{
+			preg_match('/Location:(.*?)\n/', $header, $matches);
```

Comment:
The regex `/Location:(.*?)\n/` doesn't match when the Location header is at the end.